static/script/widgets/label.js

```diff
@@ -19,151 +19,206 @@
+         * @name antie.widgets.Label
+         * @class
+         * @extends antie.widgets.Widget
+         * @param {String} [id] The unique ID of the widget. If excluded, a temporary internal ID will be used (but not included in any output).
+         * @param {String} [text] The text content of this label.
+         */
+        var Label = Widget.extend(/** @lends antie.widgets.Label.prototype */ {
+            /**
+             * @constructor
+             * @ignore
+             */
+            init: function(id, text) {
                 // The current API states that if only one parameter is passed to
                 // use that value as the text and auto generate an internal id
-				if(arguments.length == 1) {
-					this._text = id;
-					this._super();
-				} else {
-					this._text = text;
-					this._super(id);
-				}
-				this._truncationMode = Label.TRUNCATION_MODE_NONE;
-				this._maxLines = 0;
-				this._width = 0;
-				this.addClass('label');
-			},
-			/**
-			 * Renders the widget and any child widgets to device-specific output.
-			 * @param {antie.devices.Device} device The device to render to.
-			 * @returns A device-specific object that represents the widget as displayed on the device (in a browser, a DOMElement);
-			 */
-			render: function(device) {
-				// TODO: is there a more efficient way of doing this?
-				var s;
-				if(this._width && this._maxLines && this._text && (this._truncationMode == Label.TRUNCATION_MODE_RIGHT_ELLIPSIS)) {
-					var h = device.getTextHeight("fW", this._width, this.getClasses());
-					var allowedHeight = h * this._maxLines;
-					var currentHeight = device.getTextHeight(this._text, this._width, this.getClasses());
-				
-					var len = this._text.length;
-					while(currentHeight > allowedHeight && len > 1) {
-						len = Math.floor((len * allowedHeight) / currentHeight);
-						currentHeight = device.getTextHeight(this._text.substring(0, len) + "...", this._width, this.getClasses());
-					}
-					while(currentHeight <= allowedHeight && len <= this._text.length) {
-						len++;
-						currentHeight = device.getTextHeight(this._text.substring(0, len) + "...", this._width, this.getClasses());
-					}
-					len--;
-				
-					if(len < this._text.length) {
-						// truncate at word boundary
-						var boundaryLen = len;
-						while(boundaryLen && !/\w\W$/.test(this._text.substring(0, boundaryLen+1))) {
-							boundaryLen--;
-						}
-						if(boundaryLen > 0) {
-							s = this._text.substring(0, boundaryLen) + "...";
-						} else {
-							s = this._text.substring(0, len) + "...";
-						}
-					} else {
-						s = this._text;
-					}
-				} else {
-					s = this._text;
-				}
+                if(arguments.length == 1) {
+                    this._text = id;
+                    this._super();
+                } else {
+                    this._text = text;
+                    this._super(id);
+                }
+                this._truncationMode = Label.TRUNCATION_MODE_NONE;
```

Comment:
If we decide TruncatedLabel is only to be used if truncation is required, this should probably be set to TRUNCATION_MODE_RIGHT_ELLIPSIS.